Alicerce/Sources/Logging/Log+StringLogItemFormatter.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  Log+StringLogItemFormatter.swift
+//  Alicerce
+//
+//  Created by Meik Schutz on 07/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public extension Log {
+
+    public struct StringLogItemFormatter: LogItemFormatter {
+
+        private static let defaultFormatString = "$DHH:mm:ss.SSS$d $C$L$c $N.$F:$l - $M"
```

Comment:
I just thought about something which I think would be nicer than these custom format strings:

How about we define an `enum` with all the types of log components we want to log, together with associated types, when relevant:

e.g.:

```swift
typealias DateFormat = String

public enum Component { // assuming it's nested inside `StringLogItemFormatter`
    case label
    case message // the main message
    case thread
    case fileName
    case filePath
    case function
    case line
    case date(DateFormat)
    case timeZoneDate(DateFormat, TimeZone)
    case color([Component]) // every component inside will be colored
    case text(String) // static text
}
```
(not sure if I covered all the cases, but you get the idea 😆)

Then, the log "format" would look like an array of `Component`'s, e.g.: 

```swift
private static let defaultFormatComponents : [Component] = [.date("HH:mm:ss.SSS"),
                                                            .text(" "),
                                                            .color([.level]),
                                                            .text(" "),
                                                            .fileName,
                                                            .text("."),
                                                            .function,
                                                            .text(":"),
                                                            .line,
                                                            .text(" - "),
                                                            .message]
```

I think this brings some advantages:
1. adds type safety to the format
2. simplifies the format by not requiring begin/end format "tokens" (e.g. date, color)
3. and avoids tokenizing the format string every time we print a log message, and instead we simply iterate the array

What do you say?